rtos/rtx5/TARGET_CORTEX_M/rtx_lib.c

```diff
@@ -595,7 +595,9 @@ void *__user_perthread_libspace (void) {
 typedef void *mutex;
 
 // Initialize mutex
+#if !defined(__ARMCC_VERSION) || __ARMCC_VERSION < 6010050
```

Comment:
I've asked in https://github.com/ARM-software/CMSIS_5/pull/213#issuecomment-323306593 whether CMSIS guys have some better way of doing that.